Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp

```diff
@@ -244,9 +244,18 @@ void WebSWServerToContextConnection::fireBackgroundFetchClickEvent(ServiceWorker
 
 void WebSWServerToContextConnection::terminateWorker(ServiceWorkerIdentifier serviceWorkerIdentifier)
 {
+    protectedConnection()->protectedNetworkProcess()->protectedParentProcessConnection()->send(Messages::NetworkProcessProxy::AcquireServiceWorkerTerminationAssertion { webProcessIdentifier(), serviceWorkerIdentifier }, 0);
+
     send(Messages::WebSWContextManagerConnection::TerminateWorker(serviceWorkerIdentifier));
```

Comment:
I would tend to use an async reply. `SWContextManager::terminateWorker` takes a callback (though it is a Function, worth checking).